### DISCUSSION TITLE
Binary comptability with git repository

### DISCUSSION THREAD

**Discussion by alex-aparin** [libgit2/libgit2#6117] [Q&A]
Can libgit2 be used safely with existent git repo? Just a little example - will git repo valid after its processing of libgit2? I worry about probability that imported git repo with new commits (or with new pack files, blobs etc) from libgit2 will no longer accessed with original git. Of course it is simple example because there is different ways to work with git  I know that libgit2 proposes some features, which are not binary compatible with original git (because they use different organization on filesystem etc).

**Comment by ethomson** [comment]
Sorry about the delay, I did not see this discussion originally.  The entire purpose of libgit2 is compatibility with git - so yes, it should be able to be safely used with a git repo that was produced with the command-line git application, or jgit, or another git client.  There's broad interoperability goals across the ecosystem.
